js/spa/test/visibility_test.spec.ts

```diff
@@ -0,0 +1,90 @@
+import { test, expect } from "@self/tootils";
+
+test.describe("Visibility States", () => {
+	test("components with visible=true are visible and in DOM", async ({
+		page
+	}) => {
+		const textbox = page.locator("#test-textbox");
+		await expect(textbox).toBeVisible();
+	});
+
+	test("components with visible=false are removed from DOM", async ({
+		page
+	}) => {
+		await page
+			.locator("#visibility-radio")
+			.getByText("Not Visible (removed)")
+			.click();
+
+		const textbox = page.locator("#test-textbox");
+		await expect(textbox).not.toBeInViewport();
```

Comment:
Is this actually the correct way to test this? I think this test was passing before the latest commit which is what actually ensures these components are not in the DOM

Reply:
Yeh it was. I'll try to rework it.